src/Microsoft.AspNet.Mvc.Razor/RazorView.cs

```diff
@@ -190,12 +194,23 @@ private bool EnableInstrumentation
                 layoutPage.RenderBodyDelegate = bodyWriter.CopyTo;
                 bodyWriter = await RenderPageAsync(layoutPage, context, executeViewStart: false);
 
-                // Verify that RenderBody is called, or that RenderSection is called for all sections
-                layoutPage.EnsureBodyAndSectionsWereRendered();
+                // Verify that RenderBody is called
+                layoutPage.EnsureBodyWasRendered();
+
+                unRenderedSections.ExceptWith(layoutPage.RenderedSections);
+                unRenderedSections.UnionWith(layoutPage.PreviousSectionWriters.Keys.Except(layoutPage.RenderedSections,
```

Comment:
If you reverse the order of these operations I don't think you'd have to do `Except` twice